### PR TITLE
Added repetition for multiple declarations/steps

### PR DESCRIPTION
**Pull request by cmlsharp**
One of the things I missed about C for loops was the ability to use multiple declarations and multiple steps like so:

```
for (size_t i = 0, y = 3; i < 10; i++, y+=4) {}
```

I did the best I could at translating the syntax to cfor:

```
cfor!(let mut x = 0, let mut y = 3; x < 10; x+=1, y+=1; {});
```

### CODE REVIEW

**Comment by autohuonw** [comment]
Thanks for the pull request, and welcome! You should hear from @huonw (or someone else) soon.

File: src/lib.rs [hunk 1]
```diff
@@ -163,6 +163,20 @@
...
+//! # Handling multiple initializations and steps
+//!
+//! Like C loops, `cfor!` supports specfying multiple initializations and steps seperated by a comma.
+//!
+//! ```rust
+//! #[macro_use] extern crate cfor;
+//!
+//! fn main() {
+//!     cfor!{let mut x = 0, let mut y = x; x <= 10 && y <= 100; x += 1, y += 10; {
+//!         println!(x: {}, y: {}", x, y);
```
Comment (huonw) [inline src/lib.rs]: This needs an opening `"`.

**Comment by huonw** [comment]
Thanks!

Could you add a test or two to `tests/test.rs`? (e.g. `repeated_parts` that has a few `cfor` invocations with some with multiple inits, some with multiple steps and some with both.)

**Comment by cmlsharp** [comment]
Done

File: src/lib.rs (hunk 1 above)
Comment (huonw) [inline src/lib.rs]: The string passed to `println!` is still invalid (it is causing tests to fail).

**Comment by cmlsharp** [comment]
Sorry, had an error in documentation causing failed tests. Fixed now

**Comment by huonw** [comment]
Great, thanks.

**Comment by huonw** [comment]
(Included in 1.1.0, which I just published.)

**Pull request merged**